### DISCUSSION TITLE
Option for adding a button in AR mode

### DISCUSSION THREAD

**Discussion by rksambari** [google/model-viewer#3723] [Q&A]
### Description

Option for adding a button in model viewer AR mode, so that we can interact with the model in AR. After clicking on View in AR as soon as we get into AR mode, we want the button in AR mode.
[
<img width="158" alt="image" src="https://user-images.githubusercontent.com/99704624/185887822-3392cd3e-221a-4296-b30c-4ce7823ceb1f.png">
](url)

### Browser Affected

- [x] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [x] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
WebXR mode is ideal for this, as any child of the MV element is shown in AR as well. You can control visibility with CSS using `ar-status`. With Scene Viewer and Quick Look you can only add query params to the URL to enable what API they provide (which can at least create a simple button).

**Reply by rksambari** [reply]
Hi,
Thanks for the response.
I used WebXR and ar-status but I still didn't get the button(`ar-button2`) on AR mode, below is my code.

**Model Viewer**
`<model-viewer bounds="tight" enable-pan src="gltf/scene.gltf" ar ar-modes="scene-viewer webxr" camera-controls environment-image="neutral" shadow-intensity="1">
      <button slot="ar-button" id="ar-button">
          View in AR
      </button>
	  <button id="ar-button2">
          click me
      </button>
    </model-viewer>`

**CSS**
`model-viewer {
  width: 100%;
  height: 90%;
  background-color: #ffffff;
}
model-viewer[ar-status=" session-started"] > #ar-button2 {
    display: block;
}`

**Reply by elalish** [reply]
You're not using webxr, you're using scene viewer: `ar-modes="scene-viewer webxr"` mean only use webxr if scene viewer fails for some reason.

**Reply by rksambari** [reply]
Thank you for the response.

I'm able to get the button on AR with WebXR now.
Can we use javascript to add some onclick functionality for the button in AR mode.
I tried onclick functionality with Javascript, and it is working in 3D mode but not in AR mode. How to make it work in AR mode.

Thanks.

**Reply by elalish** [reply]
It should "just work" - consider this example, for instance: https://modelviewer.dev/examples/scenegraph/#variants

**Reply by mhk1121992** [reply]
> WebXR mode is ideal for this, as any child of the MV element is shown in AR as well. You can control visibility with CSS using `ar-status`. With Scene Viewer and Quick Look you can only add query params to the URL to enable what API they provide (which can at least create a simple button).

if I use ar-modes="webxr" the AR button that launches AR vanishes. I am trying my buttons to appear (like in color demo) to change the color of the model. I can do it in web but as soon as I click ar button, only the base model appears (default ar mode only).

**Reply by elalish** [reply]
@mhk1121992 Most often when WebXR is not available it's because you're serving over HTTP instead of HTTPS, which is required by the WebXR API.

**Reply by mhk1121992** [reply]
> @mhk1121992 Most often when WebXR is not available it's because you're serving over HTTP instead of HTTPS, which is required by the WebXR API.

thanks, it's working now. Is there a way to make "color demo" make work in iOS?

**Reply by elalish** [reply]
I think this works in iOS, if that's what you mean: https://modelviewer.dev/examples/scenegraph/#changeColor

**Reply by codepushed** [reply]
Is it possible to add buttons when the model is in AR preview?

**Reply by elalish** [reply]
@codetill5 Did you read the above discussion? Yes, but only in WebXR mode, which is Android only. In Quick Look and Scene Viewer, you can only add their API URL params to the `src` and we'll forward them.

**Reply by zelghrabi-edu** [reply]
@elalish how can we do it with API URL params?

**Reply by elalish** [reply]
Here are the SceneViewer [docs](https://developers.google.com/ar/develop/scene-viewer) and here is the [best](https://cwervo.com/writing/quicklook-web/) we've found on QuickLook. Simply append these hash or query params to your `src` URL. They don't overlap with each other, so you can safely put both.
